### ISSUE TITLE
Scatter report does not return correct TSV

### ISSUE THREAD

**Issue by Euphrasiologist** [genomehubs/genomehubs#222]
@rjchallis 

I'm trying to implement scatter, with a CLI API:

```bash
goat-cli taxon scatter -x assembly_span -y gc_percent -t Eukaryota -u
```

All good. Returns what I think it the URL should return as well:

```txt
GoaT lookup API URL:	https://goat.genomehubs.org/api/v2/report?result=taxon&includeEstimates=false&taxonomy=ncbi&report=table&rank=species&x=tax_tree%28Eukaryota%29%20AND%20assembly_span&y=gc_percent
```

But! It does not return the TSV I expect.

Try:

```bash
curl -X 'GET' "https://goat.genomehubs.org/api/v2/report?result=taxon&includeEstimates=false&summaryValues=count&taxonomy=ncbi&report=table&y=gc_percent&rank=species&x=tax_tree%28lamiales%29%20and%20assembly_span" -H "accept: text/tab-separated-values"
```

JSON looks good to me though:

```bash
curl -X 'GET' "https://goat.genomehubs.org/api/v2/report?result=taxon&includeEstimates=false&summaryValues=count&taxonomy=ncbi&report=table&y=gc_percent&rank=species&x=tax_tree%28lamiales%29%20and%20assembly_span" -H "accept: application/json"
```

I could implement something my side to parse the JSON but I presume this will want to be fixed for future?

Thought this was a good cross-over genomehubs issue.

_Originally posted by @Euphrasiologist in https://github.com/genomehubs/goat-cli/issues/15#issuecomment-1264665031_

**Comment by rjchallis** [comment]
sorry - didn't mean to ignore the original issue. Looks like the TSV output is still the histogram (both scatter and histogram reports share a lot of code). I'm not sure what would be the most useful TSV output here would be.

The json includes this histogram of data binned along the x axis, a 2d histogram of scatter data binned on the x and y axes, both of these per category (if applicable) and the raw scatter data if the scatterThreshold is set high enough. I would welcome any opinions on what the API should return in a TSV here, given all of the different options in the JSON. 

I had anticipated that for anyone wanting full control over plotting, they can grab the search table and group things into categories as they see fit. Not sure where you see goat-cli users fitting into the spectrum between wanting data for an instant plot vs complete control over data processing

**Comment by Euphrasiologist** [comment]
Not at all - I thought it might be better placed here on second thoughts.

Hmm, you are right actually. Doesn't seem as straight forward as I thought. The most intuitive thing might be a long (tidy?) format data of this:

```
https://goat.genomehubs.org/report?result=taxon&includeEstimates=false&summaryValues=count&taxonomy=ncbi&report=table&y=gc_percent&rank=species&x=tax_tree%28lamiales%29%20and%20assembly_span
```

What do you think? So you'd end up with three columns where the x/y were the bins and then the value?

**Comment by rjchallis** [comment]
Yes, that makes sense. Seems logical to extend this to a 4 column tidy data that includes cat as well. Table doesn't currently support x+y+cat because I wasn't sure how to present a table with 3 dimensions, but I should revisit that as the tidy data version would make a lot of sense

**Comment by Euphrasiologist** [comment]
Brilliant, sounds good. I'll implement `Scatterplot` when you get around to adding this in! :)
